app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  has_one :cart, dependent: :destroy
+  has_many :orders
+
+  after_create :create_cart
```

Comment:
oh boy, what momma said about callbacks in model? 
luckily for you this line will disappear after cart changes, but what the fuck actually :)